pinot-core/src/main/java/org/apache/pinot/core/common/datablock/DataBlockBuilder.java

```diff
@@ -289,7 +304,7 @@ public static ColumnarDataBlock buildFromColumns(List<Object[]> columns, DataSch
   }
 
   private static RowDataBlock buildRowBlock(DataBlockBuilder builder) {
-    return new RowDataBlock(builder._numRows, builder._dataSchema, builder._reverseDictionaryMap,
+    return new DataTableImplV4(builder._numRows, builder._dataSchema, builder._reverseDictionaryMap,
```

Comment:
(minor) Should this be
```suggestion
    return new RowDataBlock(builder._numRows, builder._dataSchema, builder._reverseDictionaryMap,
```

Reply:
One issue here is that `getDataBlockVersionType()` is implemented differently between `RowDataBlock` (returns 1 now) and `DataTableImplV4` (returns 4). As a result the serialized version returned by `toBytes()` is different between the two. `DataTableFactory.getDataTable()` looks for version 4 explicitly to build `DataTableImplV4`. Please check this line in testDataBlockBuilderV4() test:
`    BaseDataBlock dataBlock = (BaseDataBlock) DataTableFactory.getDataTable(rowDataBlock.toBytes());`

Reply:
I think the issue is from using `DataBlockBuilder` to build data table in the test. If we switch to use `DataTableBuilder` to build data table, it should work. Changing this can break the data block code. cc @walterddr